packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -345,8 +345,17 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
       return;
     }
 
-    const principal = this.parsePermissionPrincipal(permission.principal);
-    const { sourceAccount, sourceArn } = this.parseConditions(permission.principal) ?? {};
+    let principal = this.parsePermissionPrincipal(permission.principal);
+
+    let { sourceArn, sourceAccount, principalOrgID } = this.validateConditionCombinations(permission.principal) ?? {};
+
+    // Lambda does not actually accept an organization principal here.
+    // so the principal becomes '*' and the organization ID will be sent to
+    // the 'principalOrgID' property.
+    if (permission.principal instanceof iam.OrganizationPrincipal) {
```

Comment:
No `instanceof`. It's not safe.

The better thing to do here would be to render the principal (to `principalJson` and `conditions`) and parse those, I think...

Reply:
Removed `instanceOf`. By "it's not safe" are you primarily concerned with how `instanceof` interacts with inheritance? or something else.

Reply:
How it interacts with multiple copies of the library installed (easily happens when users are symlinking between their own packages)